### PR TITLE
Fix some map docstrings

### PR DESCRIPTION
**Pull request by dstansby**
Especially `GenericMap`, whose parameter doc seemed to be not very good previously.

### CODE REVIEW

File: sunpy/map/mapbase.py [hunk 1]
```diff
@@ -823,7 +830,7 @@ def world_to_pixel(self, coordinate, origin=0):
     @deprecated("0.8.0", alternative="sunpy.map.GenericMap.world_to_pixel")
     def data_to_pixel(self, coordinate, origin=0):
         """
-        See `~sunpy.map.mapbase.GenericMap.world_to_pixel`
```
Comment (dpshelio) [inline sunpy/map/mapbase.py]: Why a `,`?

File: sunpy/map/mapbase.py [hunk 2]
```diff
@@ -166,7 +159,21 @@ class GenericMap(NDData):
...
+        header : dict
+            A dictionary of the original image header tags.
+        plot_settings : dict, optional
+            Plot settings.
+
+        Other Parameters
+        ----------------
+        **kwargs :
+            Additional keyword arguments are passed to `NDData` init.
+        """
```
Comment (dpshelio) [inline sunpy/map/mapbase.py]: I agree with the change from `meta` to `header`, but why to put it inside the `__init__`? All the other information is in the class definition. Using your suggested changes, I don't get the help about the class when doing:
`mapbase.GenericMap?` in ipython or using `help()`

File: sunpy/map/mapbase.py [hunk 3]
```diff
@@ -166,7 +159,21 @@ class GenericMap(NDData):
...
+            A 2d list or ndarray containing the map data.
+        header : dict
+            A dictionary of the original image header tags.
+        plot_settings : dict, optional
+            Plot settings.
+
+        Other Parameters
+        ----------------
+        **kwargs :
+            Additional keyword arguments are passed to `NDData` init.
```
Comment (Cadair) [inline sunpy/map/mapbase.py]: While we are here, can you replace
```
`NDData`
```
with
```
`~astropy.nddata.NDData`
```

File: sunpy/map/mapbase.py [hunk 4]
```diff
@@ -166,7 +159,21 @@ class GenericMap(NDData):
     """
 
     def __init__(self, data, header, plot_settings=None, **kwargs):
-
+        """
+        Parameters
+        ----------
+        data : `~numpy.ndarray`, list
+            A 2d list or ndarray containing the map data.
+        header : dict
```
Comment (Cadair) [inline sunpy/map/mapbase.py]: can you add backticks around dict.
```
`dict`
```

File: sunpy/map/mapbase.py [hunk 5]
```diff
@@ -166,7 +159,21 @@ class GenericMap(NDData):
...
+            A dictionary of the original image header tags.
+        plot_settings : dict, optional
```
Comment (Cadair) [inline sunpy/map/mapbase.py]: same here.

File: sunpy/map/mapbase.py (hunk 2 above)
Comment (Cadair) [reply sunpy/map/mapbase.py]: Yeah, the parameter documentation should be in the class docstring, although I am struggling to find a reference for this convention.

Comment (dstansby) [reply sunpy/map/mapbase.py]: This seems to agree with you: 

>  Use the same sections as outlined above (all except Returns are applicable). The constructor (__init__) should also be documented here, the Parameters section of the docstring details the constructors parameters.

(https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt#class-docstring)

File: sunpy/map/mapbase.py (hunk 4 above)
Comment (dstansby) [reply sunpy/map/mapbase.py]: The objects automatically get linked here in numpydoc strings without the backticks if that's what you were thinking?

File: sunpy/map/mapbase.py (hunk 1 above)
Comment (dstansby) [reply sunpy/map/mapbase.py]: Woops!

**Comment by nabobalis** [comment]
Looks like a timeout on travis for the online tests.

**Pull request merged**